target/bin/rspamd-dkim

```diff
@@ -0,0 +1,295 @@
+  else
+    local BASE_FILE_NAME="${BASE_DIR}/${KEYTYPE}-${SELECTOR}-${DOMAIN}"
+    KEYTYPE_OPTIONS=('-t' "${KEYTYPE}")
+    _log 'info' "Creating DKIM keys of type '${KEYTYPE}' with selector '${SELECTOR}' for domain '${DOMAIN}'"
+  fi
+
+  PUBLIC_KEY_FILE="${BASE_FILE_NAME}.public.txt"
+  PUBLIC_KEY_DNS_FILE="${BASE_FILE_NAME}.public.dns.txt"
+  PRIVATE_KEY_FILE="${BASE_FILE_NAME}.private.txt"
+
+  mkdir -p "${BASE_DIR}"
+  chown _rspamd:_rspamd "${BASE_DIR}"
+
+  # shellcheck disable=SC2310
+  if __do_as_rspamd_user rspamadm \
+      dkim_keygen                 \
+      -s "${SELECTOR}"            \
+      -d "${DOMAIN}"              \
+      "${KEYTYPE_OPTIONS[@]}"     \
+      -k "${PRIVATE_KEY_FILE}"    \
+      >"${PUBLIC_KEY_FILE}"
+  then
+      _log 'info' 'Successfully created DKIM keys'
+      _log 'debug' "Public key written to '${PUBLIC_KEY_FILE}'"
+      _log 'debug' "Private key written to '${PRIVATE_KEY_FILE}'"
+  else
+    _exit_with_error 'Creating keys failed'
+  fi
+}
+
+function _check_permissions
+{
+  # shellcheck disable=SC2310
+  if ! __do_as_rspamd_user ls "${BASE_DIR}" >/dev/null
+  then
+    _log 'warn' "The Rspamd user ('_rspamd') seems to be unable to list files in the keys directory ('${BASE_DIR}') - Rspamd may experience permission errors later"
+  elif ! __do_as_rspamd_user cat "${PRIVATE_KEY_FILE}" >/dev/null
+  then
+    _log 'warn' "The Rspamd user ('_rspamd') seems to be unable to read the private key file - Rspamd may experience permission errors later"
+  else
+    _log 'debug' 'Permissions on files and directories seem ok'
+  fi
+}
+
+function _setup_default_signing_conf
+{
+  local DEFAULT_CONFIG_FILE='/etc/rspamd/override.d/dkim_signing.conf'
+  if [[ -f ${DEFAULT_CONFIG_FILE} ]]
+  then
+    _log 'debug' "'${DEFAULT_CONFIG_FILE}' exists, not supplying a default"
+  else
+    _log 'info' "Supplying a default configuration ('${DEFAULT_CONFIG_FILE}')"
+    cat >"${DEFAULT_CONFIG_FILE}" << EOF
+# documentation: https://rspamd.com/doc/modules/dkim_signing.html
+
+enabled = true;
+
+sign_authenticated = true;
+sign_local = true;
+
+use_domain = "header";
+use_redis = false;   # don't change unless Redis also provides the DKIM keys
+use_esld = true;
+
+check_pubkey = true; # you wan't to use this in the beginning
+
+domain {
+    ${DOMAIN} {
+        path = "${PRIVATE_KEY_FILE}";
+        selector = "${SELECTOR}";
+    }
+}
+
+EOF
+    chown _rspamd:_rspamd "${DEFAULT_CONFIG_FILE}"
+  fi
+}
+
+function _transform_public_key_file_to_dns_record_contents
+{
+  _log 'trace' 'Transforming DNS zone format to DNS record content now'
+  : >"${PUBLIC_KEY_DNS_FILE}"
+  grep -o '".*"' "${PUBLIC_KEY_FILE}" | tr -d '"\n' >>"${PUBLIC_KEY_DNS_FILE}"
+  printf '\n' >>"${PUBLIC_KEY_DNS_FILE}"
+
+  _log 'info' "Here is the content of the TXT DNS record ${SELECTOR}._domainkey.${DOMAIN} that you need to create:\n"
+  if ! _log_level_is '(warn|error)'
+  then
+    cat "${PUBLIC_KEY_DNS_FILE}"
+    printf '\n'
+  fi
```

Comment:
```suggestion
  if ! _log_level_is '(warn|error)'
  then
    _log 'info' "Here is the content of the TXT DNS record ${SELECTOR}._domainkey.${DOMAIN} that you need to create:\n"
    cat "${PUBLIC_KEY_DNS_FILE}"
    printf '\n'
  fi
```

Reply:
Due to the way that `_log` works, it doesn't actually matter. But your version should be a tat more efficient :D 🙈